### PR TITLE
Parser improvements, next take

### PR DESCRIPTION
**Pull request by grigory-rechistov**
* Fixed the annoying testing failures because of newlines on Windows.
* Mnemonic to iclass mangling is reworked after the discussion: instead of using hardcoded lists of iclasses, it conversion of different mnemonics to iclasses and uses a new enum value if it was successful.
* Recognition of whether a literal constant is an immediate or a relative branch is done by inspecting all operands of potentially matching iclass candidates, thus avoiding hard coded lists as well.
* Minor fixes and improvements

### CODE REVIEW

File: examples/xed-asmparse-main.c [hunk 1]
```diff
@@ -481,10 +489,46 @@ static void process_operand(xed_enc_line_parsed_t* v,
...
+        xed_uint_t offset_bits = get_constant_width(q->farptr.offset,
+                                                    q->farptr.offset_value);
+
+        seg_bits = seg_bits < 16 ? 16 : seg_bits;
+        if (seg_bits != 16) {
+            asp_error_printf(
+                "Segment value in far pointer must be 16 bits\n");
+            exit(1);
+        }
+        if (offset_bits > 32) {
```
Comment (markcharney) [inline examples/xed-asmparse-main.c]: On Intel64, far pointers in 64b mode can have 64b offsets (making for a 2+8 byte payload).

File: examples/xed-asmparse.c [hunk 2]
```diff
@@ -444,6 +445,35 @@ static int isdecorator(char* s) {
...
+static int islongptr(char *s) {
+    if (!s)
+        return 0;
+    /* skip optional "far" */
+    if (s[0] == 'F' && s[1] == 'A' && s[2] == 'R') {
+        s += 3;
+        s += skip_spaces(s, 0);
+    }
+    
+    int column_pos = -1;
```
Comment (markcharney) [inline examples/xed-asmparse.c]: probably meant to name this "colon_pos"?

**Comment by markcharney** [comment]
The commits in this pull req are included in #191.

**Comment by grigory-rechistov** [comment]
Thanks a lot!

**Pull request closed without merging**